src/main/java/org/testng/internal/TestMethodWorker.java

```diff
@@ -53,13 +55,22 @@ public TestMethodWorker(IInvoker invoker,
                           List<IClassListener> listeners)
   {
     m_invoker = invoker;
-    m_methodInstances = testMethods;
     m_suite = suite;
     m_parameters = parameters;
     m_groupMethods = groupMethods;
     m_classMethodMap = classMethodMap;
     m_testContext = testContext;
     m_listeners = listeners;
+    List<IMethodInstance> localMethodInstances = Lists.newArrayList(testMethods);
```

Comment:
@cbeust - Sure I will do that. But I would like to see if #1345 fixes this problem without needing this PR. So I am waiting for #1345 to be merged before moving forward with this PR.